central/graphql/resolvers/image_components.go

```diff
@@ -0,0 +1,115 @@
+package resolvers
+
+import (
+	"context"
+	"time"
+
+	"github.com/graph-gophers/graphql-go"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/central/metrics"
+	"github.com/stackrox/rox/pkg/features"
+	pkgMetrics "github.com/stackrox/rox/pkg/metrics"
+	"github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/utils"
+)
+
+func init() {
+	schema := getBuilder()
+	utils.Must(
+		// NOTE: This list is and should remain alphabetically ordered
+		schema.AddExtraResolvers("ImageComponent", []string{
+			"activeState(query: String): ActiveState",
+			"deploymentCount(query: String, scopeQuery: String): Int!",
+			"deployments(query: String, scopeQuery: String, pagination: Pagination): [Deployment!]!",
+			"fixedIn: String!",
+			"imageCount(query: String, scopeQuery: String): Int!",
+			"images(query: String, scopeQuery: String, pagination: Pagination): [Image!]!",
+			"imageVulnerabilityCount(query: String, scopeQuery: String): Int!",
+			"imageVulnerabilityCounter(query: String): VulnerabilityCounter!",
+			"imageVulnerabilities(query: String, scopeQuery: String, pagination: Pagination): [ImageVulnerability]!",
+			"lastScanned: Time",
+			"location(query: String): String!",
+			"topImageVulnerability: ImageVulnerability",
+			"unusedVarSink(query: String): Int",
+
+			"plottedVulns(query: String): PlottedVulnerabilities!", // TODO
+		}),
+		schema.AddQuery("imageComponent(id: ID): ImageComponent"),
+		schema.AddQuery("imageComponents(query: String, scopeQuery: String, pagination: Pagination): [ImageComponent!]!"),
+		schema.AddQuery("imageComponentCount(query: String): Int!"),
+
+		// TODO
+		schema.AddExtraResolver("ImageScan", `components(query: String, pagination: Pagination): [EmbeddedImageScanComponent!]!`),
+		schema.AddExtraResolver("ImageScan", `componentCount(query: String): Int!`),
+	)
+}
+
+// ImageComponentResolver represents a generic resolver of image component fields.
+// Values may come from either an embedded component context, or a top level component context.
+// NOTE: This list is and should remain alphabetically ordered
+type ImageComponentResolver interface {
+	ActiveState(ctx context.Context, args RawQuery) (*activeStateResolver, error)
+	DeploymentCount(ctx context.Context, args RawQuery) (int32, error)
+	Deployments(ctx context.Context, args PaginatedQuery) ([]*deploymentResolver, error)
+	FixedIn(ctx context.Context) string
+	FixedBy(ctx context.Context) string
```

Comment:
These seem redundant. We can come back to it in follow-up. Create a tracking JIRA.

Reply:
Makes sense. We should deprecate `FixedIn`. Let's create a JIRA to track deprecation on `FixedIn`. The field is being used by FE, therefore, we should make sure we move to the auto-generated field without breaking the FE.

Reply:
I marked FixedIn as deprecated in the graphql for the time being, here is the JIRA tracker (note not in the scope of this epic):

https://issues.redhat.com/browse/ROX-11328